### DISCUSSION TITLE
Make setting up and getting started with the project much easier

### DISCUSSION THREAD

**Discussion by sonalranjit** [bikespace/bikespace#197] [Onboarding & Getting Started]
Setting up the project can be simplified further. Such as locking down versions of node and python and other dependencies required for the project.
- Check for the proper `node` version, preferably a LTS version. (Currently it's v18.16.0)
- Check for the proper `python` version. (Currently python3.9)
- Check to see if `gatsby-cli` is installed.
- Check to see if postgres is running locally with a database called `bikespace_dev`
- Update README to reflect these changes and setup processes

**Comment by mding5692** [comment]
One thing to note is also that `PYTHON_VERSION` is specified as 3.9 so would get `python3.9 -m venv...` which can be issue if need to update to most recent version of python and if they just use python (without version number) as their python command on command line, see:
https://github.com/bikespace/bikespace/blob/main/Makefile#L28

I edited Makefile to make it work on my computer but this place might need to be revisited or updated

Oh this is Michael by the way

**Comment by tallcoleman** [comment]
@sonalranjit did you fix most of these with the recent README update etc.?

**Comment by hulloanson** [comment]
@mding5692 update readme to tell developer about the minimum python version

**Comment by sonalranjit** [comment]
I've just added the check for python version and updated the readme to mention minimum python version of 3.9.6 with this PR https://github.com/bikespace/bikespace/pull/118 

the other checks for gatsby and node are not automated as part of the make commands 😬

**Comment by tallcoleman** [comment]
@sonalranjit I figured out why I was having trouble with `make`, it seems you can't have any spaces in your filepaths at all. One of the parent folders where I have the bikespace repo had a space in it, which is why make was failing out every time I tried to run one of the commands.

**Comment by AidanNPH** [comment]
I am happy to work on this one

**Comment by hulloanson** [comment]
@AidanNPH is going to try setting up a local development env and update the README and Makefile.

Ideally the process should be a one-liner, but a few commands written in the README could work as well

**Comment by tallcoleman** [comment]
Decided to try setting this up on our windows desktop - some notes:

Looked up some options to run `make` on windows and decided to try using the [Windows Subsystem for Linux](https://learn.microsoft.com/en-us/windows/wsl/) (WSL) which worked pretty well. (Especially with the [VS Code integration](https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-vscode)) This may have also created and/or solved other problems down the line, not sure...

## bikespace_frontend (develop branch)

In WSL, installed:

* `make` (I think I just ran `sudo apt install make` or whatever it recommended after trying to run `make` unsuccessfully)
* [`nvm`](https://github.com/nvm-sh/nvm?tab=readme-ov-file) using the recommended curl command
* [`node`](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) using `nvm install node`

After that, `make dev-frontend` worked. A little slow, but I was running it from a windows directory, [which is not the recommended way](https://learn.microsoft.com/en-us/windows/wsl/filesystems#file-storage-and-performance-across-file-systems)

## bikespace_api

### Setting up Postgres
In WSL ran:
* `sudo apt install postgresql`
* `sudo -i -u postgres` (switch to postgres user)
* `createdb bikespace_dev`
* `psql`
* `ALTER USER postgres PASSWORD 'postgres';`
* `quit` (to close psql)
* `exit` (to switch back from postgres user)

This was a helpful reference for some of the commands: https://www.digitalocean.com/community/tutorials/how-to-install-postgresql-on-ubuntu-20-04-quickstart#step-4-creating-a-new-database

### Getting Python to work

I followed the instructions here to install python 3.12 on Ubuntu:
https://ubuntuhandbook.org/index.php/2023/05/install-python-3-12-ubuntu/

To get python 3.12 as the default for the `python3` command, I modified the instructions from the page above and also installed `venv`:

`sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.12 1`
`sudo update-alternatives --config python3`
`sudo apt install python3.12-venv`

### Setting up the database and running the API

From there, the instructions in the readme worked (i.e. run `make recreate-db`, then `make seed-db`, then `make run-flask-app`

**Comment by tallcoleman** [comment]
One contributor reported issues getting playwright working on gentoo linux
